File: Data Structures & Algorithms/binary-tree-right-side-view/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        
        List<Integer> ans = new ArrayList<>();
        if(root == null){
            return ans;
        }

        Deque<TreeNode> dq = new ArrayDeque<>();
        dq.offer(root);

        while(!dq.isEmpty()){

            int rightMost=0;
            int size = dq.size();
            for(int i=0;i<size;i++){
                TreeNode curr = dq.poll();
                rightMost = curr.val;
                if(curr.left != null){
                    dq.offer(curr.left);
                }
                if(curr.right != null){
                    dq.offer(curr.right);
                }

            }
            ans.add(rightMost);
        }
        return ans;
    }
}

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        
        if not root:
            return 0

        stack = [(root,root.val)]
        count = 0

        while stack:
            currNode,maxSofar = stack.pop()

            if(currNode.val >= maxSofar):
                count +=1
            
            maxSofar = max(maxSofar,currNode.val)

            if currNode.right:
                stack.append((currNode.right,maxSofar))
            if currNode.left:
                stack.append((currNode.left,maxSofar))
        return count;
        